field_utils/tests/Test_FieldBundleDelta.pf

```diff
@@ -0,0 +1,530 @@
+      call MAPL_InfoSetInternal(field, KEY_UNGRIDDED_DIMS, value=ungridded_info)
+
+      call ESMF_FieldEmptyComplete(field, typekind=typekind, ungriddedLBound=bounds%lower, ungriddedUBound=bounds%upper)
+      call FieldSet(field, FILL_VALUE)
+
+   end subroutine setup_field
+
+   subroutine teardown_field(field)
+      type(ESMF_Field), intent(inout) :: field
+
+      call ESMF_FieldDestroy(field)
+
+   end subroutine teardown_field
+
+   subroutine setup_bundle(bundle, weights, geom, typekind, units, with_ungridded)
+      type(ESMF_FieldBundle), intent(out) :: bundle
+      real(kind=ESMF_KIND_R4), intent(in) :: weights(:)
+      type(ESMF_Geom), intent(in) :: geom
+      type(ESMF_TypeKind_Flag), intent(in) :: typekind
+      character(len=*), intent(in) :: units
+      logical, optional, intent(in) :: with_ungridded
+      
+      integer :: i
+      type(ESMF_Field) :: f
+      integer :: fieldCount
+      type(UngriddedDims) :: ungridded_dims
+      type(ESMF_Info) :: ungridded_info
+
+      bundle = ESMF_FieldBundleCreate()
+      call MAPL_FieldBundleSet(bundle, geom=geom)
+      fieldCount = size(weights) - 1
+      do i = 1, fieldCount
+         call setup_field(f, geom, typekind, units, with_ungridded=with_ungridded)
+         call ESMF_FieldBundleAdd(bundle, [f], multiflag=.true.)
+      end do
+
+      ungridded_dims = UngriddedDims()
+      ungridded_info = ungridded_dims%make_info()
+      call MAPL_InfoSetInternal(bundle, KEY_UNGRIDDED_DIMS, value=ungridded_info)
+
+      call MAPL_InfoSetInternal(bundle, KEY_INTERPOLATION_WEIGHTS, weights)
+      if (typekind == ESMF_TYPEKIND_R4) then
+         call MAPL_InfoSetInternal(bundle, KEY_TYPEKIND, "R4")
+      else
+         call MAPL_InfoSetInternal(bundle, KEY_TYPEKIND, "R8")
+      end if
+      call MAPL_InfoSetInternal(bundle, KEY_UNITS, units)
+
+      call MAPL_InfoSetInternal(bundle, KEY_VLOC, "VERTICAL_DIM_NONE")
+      ungridded_dims = UngriddedDims()
+
+      if (present(with_ungridded)) then
+         if (with_ungridded) then
+            call MAPL_InfoSetInternal(bundle, KEY_VLOC, "VERTICAL_DIM_CENTER")
+            call MAPL_InfoSetInternal(bundle, KEY_NUM_LEVELS, NUM_LEVELS)
+            call ungridded_dims%add_dim(UngriddedDim(NUM_RADII, "radius", 'nm'))
+         end if
+      end if
+
+      ungridded_info = ungridded_dims%make_info()
+      call MAPL_InfoSetInternal(bundle, KEY_UNGRIDDED_DIMS, value=ungridded_info)
+
+   end subroutine setup_bundle
+
+   subroutine teardown_bundle(bundle)
+      type(ESMF_FieldBundle), intent(inout) :: bundle
+      type(ESMF_Field), allocatable :: fieldList(:)
+
+      integer :: i
+
+      call MAPL_FieldBundleGet(bundle, fieldList=fieldList)
+      do i = 1, size(fieldList)
+         call ESMF_FieldDestroy(fieldList(i))
+      end do
+      call ESMF_FieldBundleDestroy(bundle)
+
+   end subroutine teardown_bundle
+
+   @test(type=ESMF_TestMethod, npes=[1])
+   subroutine test_change_typekind(this)
```

Comment:
Does this check if the typekind has changed in the bundle?

Reply:
Yes - albeit indirectly by retrieving a pointer to an R8 array.